be/src/storage/rowset/map_column_iterator.cpp

```diff
@@ -39,6 +37,26 @@ Status MapColumnIterator::init(const ColumnIteratorOptions& opts) {
     RETURN_IF_ERROR(_offsets->init(opts));
     RETURN_IF_ERROR(_keys->init(opts));
     RETURN_IF_ERROR(_values->init(opts));
+
+    if (_paths.empty()) {
+        _access_keys = true;
+        _access_values = true;
+        return Status::OK();
+    }
+
+    _access_keys = false;
+    _access_values = false;
+
+    for (const auto& p : _paths) {
+        if (p->is_key()) {
+            _access_keys |= true;
+        }
+
+        if (p->is_offset()) {
+            _access_keys |= false;
```

Comment:
Is it works?

Reply:
`map_size()` only read offset, may be we don't need the check